src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyConsoleLogger.cs

```diff
@@ -101,9 +101,8 @@ private void WriteMessage(LogLevel logLevel, string logName, int eventId, string
                         case LogLevel.Critical:
                             _jsRuntime.InvokeUnmarshalled<string, object>("Blazor._internal.dotNetCriticalError", formattedMessage);
                             break;
-                        default: // LogLevel.None or invalid enum values
-                            Console.WriteLine(formattedMessage);
-                            break;
+                        default: // invalid enum values
+                            throw new ArgumentException(nameof(logLevel));
```

Comment:
The parameter to ArgumentException's ctor is a message, not a parameter name:

https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception.-ctor?view=net-5.0#System_ArgumentException__ctor_System_String_

There is an overload that takes both, though (but get the order right 😄 ):

https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception.-ctor?view=net-5.0#System_ArgumentException__ctor_System_String_System_String_

Reply:
Also, this is now throwing an exception where it didn't used to, so perhaps a breaking change (in a super obscure case)? Maybe just call jsRuntime w/ console.error(...) as seen in earlier lines?

Reply:
> The parameter to ArgumentException's ctor is a message, not a parameter name:

(doh) I swear I've been bit by this too many times.... Thanks for the catch.

> Also, this is now throwing an exception where it didn't used to, so perhaps a breaking change

It's for the case where someone passes in an invalid `LogLevel` enum into `Log`. Is changing this to an exception not acceptable? I would think devs who are casting random ints to `LogLevel` would want to know they have a bug in their code.

We throw for other loggers:

https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs#L101
https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/libraries/Microsoft.Extensions.Logging.Console/src/SystemdConsoleFormatter.cs#L115
https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/libraries/Microsoft.Extensions.Logging.Console/src/SimpleConsoleFormatter.cs#L152

Reply:
No idea whether it's acceptable, but it's a change. I agree it's super obscure. I'll let one of the code owners help answer that as I am not a stakeholder.

Reply:
@pranavkm @SteveSandersonMS - any thoughts here? Is changing this to an exception acceptable?

Reply:
(BTW the ArgNull parameter value still needs to be fixed.)

Reply:
Agreed - but I only want to fix it once. If we can't even throw an exception here, I won't bother an intermediate change.

Reply:
@captainsafia are you able to look at this change? IMO, throwing for an unsupported log level does not seem unreasonable.